vendor/src/github.com/docker/engine-api/types/container/config.go

```diff
@@ -5,6 +5,27 @@ import (
 	"github.com/docker/go-connections/nat"
 )
 
+// HealthConfig holds configuration settings for the HEALTHCHECK feature.
+type HealthConfig struct {
+	// Test to perform to check container is healthy.
+	// An empty slice means to inherit the default.
+	// Otherwise, Cmd[0] is "NONE" or "CMD".
+	Test strslice.StrSlice `json:",omitempty"`
+
+	// Nil means to inherit.
+	GracePeriod *float64 `json:",omitempty"` // Time after container start to assume failure means still starting.
+	Interval    *float64 `json:",omitempty"` // Time to wait between checks.
+	Timeout     *float64 `json:",omitempty"` // Time to wait before considering the check to have hung.
+
+	// Number of consecutive failures needed to consider a container as unhealthy.
+	// Zero means inherit.
+	Retries uint `json:",omitempty"`
+
+	// Stop container on entering unhealthy state.
+	// Nil means inherit.
+	ExitOnUnhealthy *bool `json:",omitempty"`
```

Comment:
If you lose the pointer then you have to throw away one of the current options:

- Use policy specified in the image (`docker run image`).
- Exit on unhealthy, whatever the image says (`docker run --exit-on-unhealthy image`).
- Do not exit on unhealthy, whatever the image says (`docker run --exit-on-unhealthy=false image`).

Which are you proposing to drop?